File: consultoria/src/main/java/br/com/sigo/consultoria/enums/EnumAtributosPesquisaEmpresa.java
package br.com.sigo.consultoria.enums;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public enum EnumAtributosPesquisaEmpresa {

  @JsonProperty("nome")
  NOME("nome", "nome_empresa"),
  @JsonProperty("id")
  ID("id", "id"),
  @JsonProperty("cnpj")
  CNPJ("cnpj", "cnpj"),
  ;

  private String param;

  private String atributo;

  public String getParam() {
    return param;
  }

  public String getAtributo() {
    return atributo;
  }
}

File: consultoria-app/app/src/main/java/br/com/sigo/consultoria/enums/EnumAtributosPesquisaEmpresa.kt
package br.com.sigo.consultoria.enums

enum class EnumAtributosPesquisaEmpresa(
    val param: String,
    val atributo: String
) {
    NOME("nome", "nome_empresa"), ID("id", "id"), CNPJ("cnpj", "cnpj");

}